snapcraft/commands/lxc

```diff
@@ -2,8 +2,11 @@
 set -eu
 
 # Re-exec outside of apparmor confinement
-if [ -d /sys/kernel/security/apparmor ] && [ "$(while read -r l; do echo "$l"; done < /proc/self/attr/current)" != "unconfined" ]; then
-    exec /usr/bin/aa-exec -p unconfined -- "$0" "$@"
+if [ -d /sys/kernel/security/apparmor ]; then
+  label="$(while read -r l; do echo "$l"; done < /proc/self/attr/current)"
+  if [ "$label" != "unconfined" ] && [ -n "${label##*(unconfined)}" ]; then
+    exec aa-exec -p unconfined -- "$0" "$@"
```

Comment:
```suggestion
    exec /usr/bin/aa-exec -p unconfined -- "$0" "$@"
```

This was added to avoid PATH lookup overhead. I don't know how big of a difference it makes but might as well keep it, if you don't mind.